stdlib/src/collections/_parsing_numbers/parsing_floats.mojo

```diff
@@ -0,0 +1,341 @@
+                + String(input_string)
+                + "'"
+            )
+
+    if not dot_or_e_found:
+        # We were reading the significand
+        significand = exponent
+        exponent = InlineArray[UInt8, size=std_size](ord("0"))
+
+    exponent_as_integer = (
+        exponent_multiplier * to_integer(exponent) - additional_exponent
+    )
+    significand_as_integer = to_integer(significand)
+    return (significand_as_integer, exponent_as_integer.cast[DType.int64]())
+
+
+fn strip_unused_characters(x: String) -> String:
+    result = x.strip()
+    result = result.removesuffix("f")
+    result = result.removesuffix("F")
+    result = result.removeprefix("+")
+    return result
+
+
+fn get_sign(x: String) -> Tuple[Float64, String]:
+    if x.startswith("-"):
+        return (-1.0, x[1:])
+    return (1.0, x)
+
+
+# Powers of 10 and integers below 2**53 are exactly representable as Float64.
+# Thus any operation done on them must be exact.
+fn can_use_clinger_fast_path(w: UInt64, q: Int64) -> Bool:
+    return w <= UInt64(2**53) and (Int64(-22) <= q <= Int64(22))
+
+
+fn clinger_fast_path(w: UInt64, q: Int64) -> Float64:
+    if q >= 0:
+        return w.cast[DType.float64]() * powers_of_10[int(q)]
+    else:
+        return w.cast[DType.float64]() / powers_of_10[int(-q)]
+
+
+fn full_multiplication(x: UInt64, y: UInt64) -> UInt128:
+    # Note that there are assembly instructions to
+    # do all that on some architectures.
+    # That should speed things up.
+    var x_low = x & 0xFFFFFFFF
+    var x_high = x >> 32
+    var y_low = y & 0xFFFFFFFF
+    var y_high = y >> 32
+
+    var low_low = x_low * y_low
+    var low_high = x_low * y_high
+    var high_low = x_high * y_low
+    var high_high = x_high * y_high
+
+    var carry = (low_low >> 32) + (low_high & 0xFFFFFFFF) + (
+        high_low & 0xFFFFFFFF
+    )
+
+    var low = low_low + (low_high << 32) + (high_low << 32)
+    var high = high_high + (low_high >> 32) + (high_low >> 32) + (carry >> 32)
+
+    return UInt128(high, low)
+
+
+fn get_128_bit_truncated_product(w: UInt64, q: Int64) -> UInt128:
+    alias bit_precision = mantissa_explicit_bits + 3
+    var index = 2 * (q - smallest_power_of_five)
+    var first_product = full_multiplication(w, get_power_of_5(int(index)))
+
+    var precision_mask = UInt64(0xFFFFFFFFFFFFFFFF) >> bit_precision
+    if (first_product.high & precision_mask) == precision_mask:
+        second_product = full_multiplication(w, get_power_of_5(int(index + 1)))
+        first_product.low = first_product.low + second_product.high
+        if second_product.high > first_product.low:
+            first_product.high = first_product.high + 1
+
+    return first_product
+
+
+fn create_float64(m: UInt64, p: Int64) -> Float64:
+    var m_mask = UInt64(2**52 - 1)
+    var representation_as_int = (m & m_mask) | (
+        (p + 1023).cast[DType.uint64]() << 52
+    )
+
+    return UnsafePointer.address_of(representation_as_int).bitcast[Float64]()[]
```

Comment:
Could use `memory.bitcast` here.

Reply:
Indeed, good catch! The function was well hidden!